### PR TITLE
Add check for pwned secrets to resolution center

### PR DESCRIPTION
**Pull request by pvizeli**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by pvizeli** [comment]
More info page: https://github.com/home-assistant/home-assistant.io/pull/16730

**Comment by ludeeus** [comment]
Checking this should have a Job with the `INTERNET_SYSTEM` JobCondition, and the new one in #2617

**Comment by ludeeus** [comment]
So we need to adjust this <https://www.home-assistant.io/privacy/>?

File: supervisor/resolution/checks/base.py [hunk 1]
```diff
@@ -61,6 +64,11 @@ def issue(self) -> IssueType:
     def context(self) -> ContextType:
         """Return a ContextType enum."""
 
+    @property
+    def multible(self) -> bool:
+        """Return True if they can have multible issues referenced by reference."""
```
Comment (frenck) [inline supervisor/resolution/checks/base.py:69]:
```suggestion
    def multiple(self) -> bool:
        """Return True if they can have multiple issues referenced by reference."""
```

File: supervisor/resolution/notify.py [hunk 2]
```diff
@@ -52,6 +52,14 @@ async def issue_notifications(self):
                             "notification_id": "supervisor_update_home_assistant_2021_1_5",
                         }
                     )
+            if issue.type == IssueType.PWNED and issue.context == ContextType.ADDON:
+                messages.append(
+                    {
+                        "title": f"Insecure Secrets on {issue.reference}",
+                        "message": f"The Add-on {issue.reference} use secrets which are detected as not secure, see see https://www.home-assistant.io/more-info/pwned-passwords for more information.",
```
Comment (frenck) [inline supervisor/resolution/notify.py:59]:
```suggestion
                        "message": f"The Add-on {issue.reference} uses secrets which are detected as not secure, see see https://www.home-assistant.io/more-info/pwned-passwords for more information.",
```

File: supervisor/utils/pwned.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            data = await request.text()
+
+        buffer = io.StringIO(data)
+        for line in buffer.readline():
+            if sha1_pw != line.split(":")[0]:
+                continue
+            return True
+
+    except (aiohttp.ClientError, asyncio.TimeoutError) as err:
+        _LOGGER.warning("Can't fetch freegeoip data: %s", err)
```
Comment (frenck) [inline supervisor/utils/pwned.py:31]:
```suggestion
        _LOGGER.warning("Can't fetch hibp data: %s", err)
```

File: supervisor/resolution/checks/base.py [hunk 4]
```diff
@@ -24,31 +23,35 @@ async def __call__(self) -> None:
...
+                issue.reference,
+            )
+            if await self.approve_check(reference=issue.reference):
+                continue
+            self.sys_resolution.dismiss_issue(issue)
 
-        # Check if issue still exists
-        if await self.approve_check():
+        # System is not affected
+        if affected and not self.multible:
```
Comment (frenck) [inline supervisor/resolution/checks/base.py:44]:
```suggestion
        if affected and not self.multiple:
```

**Review by frenck — APPROVED** [review]
Sweet stuff Pascal!

**Comment by frenck** [comment]
> So we need to adjust this [home-assistant.io/privacy](https://www.home-assistant.io/privacy/)?

Not sure, as the password is never sent and not collected either.

<https://www.troyhunt.com/ive-just-launched-pwned-passwords-version-2/#cloudflareprivacyandkanonymity>

**Pull request merged**